client/src/shared/components/Menu/Menu.js

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import Box from '../Box/Box';
+import Button from '../Button/Button';
+import Stack from '../Stack/Stack';
+import styles from './Menu.module.css';
+
+const Menu = ({ children, className, align, button }) => {
+  const [showMenu, setShowMenu] = useState(false);
+  let didClickButton = false;
+  useEffect(() => {
+    let timeout;
+    function handleRootClick() {
+      // The timeout will ensure this gets executed after other click handlers
```

Comment:
Interesting approach! Could also probably use the `event.target` to figure out what was clicked and decide whether to hide or show based on that if you wanted to avoid the timeout logic.

Reply:
So i updated the branch to have less hacky logic for detecting the menu button being pressed, but I still think i need the timeout (or something) to ensure the menu closes only after the click handler(s) for the child elements have a chance to resolve.